gke-gs-bucket-backup/gs_bucket_sync.py

```diff
@@ -120,49 +120,55 @@ def tar_directory(self, tar_dir: str) -> str:
     def upload_encrypted_timestamp_file(self, upload_file: str) -> None:
         """ Given a file path upload said file to our backup bucket.
             File is timestamp'd and includes file prefix. """
-        timestamp_name = datetime.datetime.now().strftime(
-            '%F-%R-%s-{}.tar.gz'.format(self.suffix))
-        backup_bucket_path = os.path.join(self.dst, timestamp_name)
+        timestamp = datetime.datetime.now().strftime('%F-%R-%s')
+        backup_bucket_path = os.path.join(self.dst, '{}-{}'.format(timestamp, self.filename))
 
         self.gsutil_encrypt_cp_cmd(upload_file, backup_bucket_path)
 
 
 if __name__ == "__main__":
     args = argparse.ArgumentParser(description=__doc__)
-    args.add_argument('src', type=str, help="Source bucket")
-    args.add_argument('backup', type=str, help="Backup bucket")
-    args.add_argument(
-        'suffix', type=str,
-        help="Backup suffix filename ex: $timestamp-prefix.tar.gz")
-    args.add_argument('-v', action='store_true', default=False,
+    args.add_argument('-v', '--verbose', action='store_true', default=False,
                       required=False, help='Increase verbosity')
-    args.add_argument('-gsutil_bin', type=str, required=False,
+    args.add_argument('-g', '--gsutil', type=str, required=False,
                       help="Full path to gsutil binary on disk",
                       default="/usr/bin/gsutil")
-    args.add_argument('-svc_act_key', type=str, required=False,
+    args.add_argument('-s', '--svc-acct-key', type=str, required=False,
                       help="Full path to a GCP service account key")
     parsed_args = args.parse_args()
 
     # Set verbose output
-    if parsed_args.v:
+    if parsed_args.verbose:
         logger.setLevel(logging.DEBUG)
 
     logger.debug("ARGS piped in: {}".format(parsed_args))
 
-    try:
-        encrypt_key = os.environ['GBACKUP_ENV_KEY']
-    except KeyError:
-        logging.error("Must expose an env var called GBACKUP_ENV_KEY")
+    backup_src = os.environ.get("GS_BACKUP_SRC")
+    backup_dest = os.environ.get("GS_BACKUP_DEST")
+    filename = os.environ.get("GS_BACKUP_FILENAME")
+    encrypt_key = os.environ.get("GS_ENCRYPTION_KEY")
+
+    if not all((backup_src, backup_dest, filename, encrypt_key)):
         args.print_help()
+        print()
+        print("Required environment variables:")
+        print("  GS_BACKUP_SRC: bucket URL prefix (i.e. gs://files.example.org/stuff)")
+        print("  GS_BACKUP_DEST: bucket URL prefix (i.e. gs://example-org-backups/files)")
+        print("  GS_BACKUP_FILENAME: object name, will be prefixed with timestamp (i.e. stuff.tar.gz)")
+        print("  GS_ENCRYPTION_KEY: key used to encrypt object")
```

Comment:
Solid approach. I like that it vets the args and bails out early. Consider using `logger.error()` rather than `print()`, though!